File: app/src/main/java/com/dof100/morsenotifier/MyPreferencesFragment.java
package com.dof100.morsenotifier;

import android.content.Intent;
import android.content.SharedPreferences;
import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
import android.os.Bundle;
import android.preference.CheckBoxPreference;
import android.preference.Preference;
import android.preference.PreferenceCategory;
import android.preference.PreferenceFragment;
import android.preference.PreferenceGroup;
import android.preference.PreferenceManager;
import android.preference.PreferenceScreen;
import java.util.ArrayList;
import java.util.Locale;

public class MyPreferencesFragment extends PreferenceFragment implements OnSharedPreferenceChangeListener {
   private void a() {
      if (App.d) {
         MyPreferenceList var1 = (MyPreferenceList)this.findPreference("pref_voice_locale");
         if (var1 == null) {
            MyLog.log("MyPreferencesFragment.init_pref_voice_locale ERROR pref_voice_locale=null");
         } else {
            C0149l var2 = App.a(this.getActivity().getApplicationContext());
            if (var2 == null) {
               MyLog.log("MyPreferencesFragment.init_pref_voice_locale ERROR mPlayerTTS=null");
            } else if (var2.b == null) {
               MyLog.log("MyPreferencesFragment.init_pref_voice_locale ERROR App.mPlayerTTS.mTTS=null");
            } else if (var2.a == -1) {
               MyLog.log("MyPreferencesFragment.init_pref_voice_locale ERROR mTTSStatus=ERROR");
            } else {
               Locale[] var3 = Locale.getAvailableLocales();
               ArrayList var4 = new ArrayList();
               int var5 = var3.length;

               for(int var6 = 0; var6 < var5; ++var6) {
                  Locale var7 = var3[var6];
                  String var8;
                  if (var7.toString().equals("en_US_POSIX")) {
                     var8 = "MyPreferencesFragment.init_pref_voice_locale Locale en_US_POSIX not supported";
                  } else {
                     int var9 = var2.b.isLanguageAvailable(var7);
                     if (var9 < 1) {
                        continue;
                     }

                     var4.add(var7.toString());
                     StringBuilder var10 = new StringBuilder();
                     var10.append("MyPreferencesFragment.init_pref_voice_locale Locale: ");
                     var10.append(var7.toString());
                     var10.append(" supported ");
                     var10.append(var9);
                     var8 = var10.toString();
                  }

                  MyLog.log(var8);
               }

               String[] var11;
               if (var4.size() > 0) {
                  var11 = (String[])var4.toArray(new String[var4.size()]);
               } else {
                  var11 = new String[]{"(none)"};
               }

               var1.setEntries(var11);
               var1.setEntryValues(var11);
               var1.a();
            }
         }
      }
   }

   private void a(Preference var1) {
      if (var1 == null) {
         MyLog.log("MyPreferencesFragment.checkAllPreferences: p=null");
      } else {
         if (var1 instanceof PreferenceCategory || var1 instanceof PreferenceScreen) {
            PreferenceGroup var2 = (PreferenceGroup)var1;

            for(int var3 = var2.getPreferenceCount() - 1; var3 >= 0; --var3) {
               this.a(var2.getPreference(var3));
            }
         }

         this.b(var1);
      }
   }

   private void b() {
      MyLog.log("MyPreferencesFragment.check_pref_voice_locale");
   }

   private void b(Preference var1) {
      if (var1 == null) {
         MyLog.log("MyPreferencesFragment.checkPreference: aP = null");
      } else {
         String var2 = var1.getKey();
         if (var2 != null) {
            if (var2.startsWith("pref_voice_locale")) {
               this.b();
            }

            if (var2.startsWith("pref_call_contact")) {
               this.c();
            } else if (var2.startsWith("pref_sms_contact")) {
               this.d();
            } else if (var2.startsWith("pref_reminders_select")) {
               this.c(var1);
            } else if (var2.startsWith("pref_apps_filters")) {
               this.d(var1);
            } else if (var2.startsWith("pref_apps_recentnotifications")) {
               this.e(var1);
            } else {
               if (var1 instanceof PreferenceCategory || var1 instanceof PreferenceScreen) {
                  StringBuilder var3;
                  String var4;
                  if (!var2.equals("pref_morse") && !var2.equals("pref_audio") && !var2.equals("pref_display")) {
                     if (!var2.equals("pref_voice") || !App.c) {
                        return;
                     }

                     var3 = new StringBuilder();
                     var4 = "MyPreferencesFragment.checkPreference: remove voice key = ";
                  } else {
                     if (!App.d) {
                        return;
                     }

                     var3 = new StringBuilder();
                     var4 = "MyPreferencesFragment.checkPreference: remove morse key = ";
                  }

                  var3.append(var4);
                  var3.append(var2);
                  MyLog.log(var3.toString());
                  var1.setEnabled(false);
               }

            }
         }
      }
   }

   private void c() {
      CheckBoxPreference var1 = (CheckBoxPreference)this.findPreference("pref_call_contactdisplayname");
      CheckBoxPreference var2 = (CheckBoxPreference)this.findPreference("pref_call_contactfirstname");
      CheckBoxPreference var3 = (CheckBoxPreference)this.findPreference("pref_call_contactlastname");
      CheckBoxPreference var4 = (CheckBoxPreference)this.findPreference("pref_call_contactinitials");
      CheckBoxPreference var5 = (CheckBoxPreference)this.findPreference("pref_call_contactnickname");
      if (var1 != null && var2 != null && var3 != null && var4 != null && var5 != null) {
         boolean var6;
         if (!var1.isChecked() && !var2.isChecked() && !var3.isChecked() && !var4.isChecked() && !var5.isChecked()) {
            var6 = false;
         } else {
            var6 = true;
         }

         this.findPreference("pref_call_contactname_none").setEnabled(var6);
      }

   }

   private void c(Preference var1) {
      MyLog.log("MyPreferencesFragment.check_pref_reminders_select");
      int var2 = PreferenceManager.getDefaultSharedPreferences(this.getActivity().getApplicationContext()).getInt("reminders_n", 0);
      MyPreferenceText var3;
      String var4;
      switch(var2) {
      case 0:
         var3 = (MyPreferenceText)var1;
         var4 = "No Reminders";
         break;
      case 1:
         var3 = (MyPreferenceText)var1;
         var4 = "1 reminder";
         break;
      default:
         var3 = (MyPreferenceText)var1;
         var4 = String.format(Locale.US, "%d reminders", var2);
      }

      var3.a(var4);
   }

   private void d() {
      CheckBoxPreference var1 = (CheckBoxPreference)this.findPreference("pref_sms_contactdisplayname");
      CheckBoxPreference var2 = (CheckBoxPreference)this.findPreference("pref_sms_contactfirstname");
      CheckBoxPreference var3 = (CheckBoxPreference)this.findPreference("pref_sms_contactlastname");
      CheckBoxPreference var4 = (CheckBoxPreference)this.findPreference("pref_sms_contactinitials");
      CheckBoxPreference var5 = (CheckBoxPreference)this.findPreference("pref_sms_contactnickname");
      if (var1 != null && var2 != null && var3 != null && var4 != null && var5 != null) {
         boolean var6;
         if (!var1.isChecked() && !var2.isChecked() && !var3.isChecked() && !var4.isChecked() && !var5.isChecked()) {
            var6 = false;
         } else {
            var6 = true;
         }

         this.findPreference("pref_sms_contactname_none").setEnabled(var6);
      }

   }

   private void d(Preference var1) {
      MyLog.log("MyPreferencesFragment.check_pref_apps_filters");
      int var2 = MyAppNotificationFilters.c(this.getActivity().getApplicationContext());
      String var3;
      MyPreferenceText var4;
      if (var2 > 1) {
         var4 = (MyPreferenceText)var1;
         var3 = String.format(Locale.US, this.getString(2131492959), var2);
      } else {
         if (var2 == 1) {
            var4 = (MyPreferenceText)var1;
            var2 = 2131492958;
         } else {
            var4 = (MyPreferenceText)var1;
            var2 = 2131492960;
         }

         var3 = this.getString(var2);
      }

      var4.a(var3);
   }

   private void e(Preference var1) {
      MyLog.log("MyPreferencesFragment.check_pref_apps_recentnotifications");
      int var2 = MyRecentAppNotifications.n(this.getActivity().getApplicationContext());
      MyPreferenceText var3;
      String var4;
      if (var2 > 1) {
         var3 = (MyPreferenceText)var1;
         var4 = String.format(Locale.US, "%d recent notifications", var2);
      } else if (var2 == 1) {
         var3 = (MyPreferenceText)var1;
         var4 = "1 recent notification";
      } else {
         var3 = (MyPreferenceText)var1;
         var4 = "No recent notifications";
      }

      var3.a(var4);
   }

   public void onActivityCreated(Bundle var1) {
      MyLog.log("MyPreferencesFragment.onActivityCreated");
      super.onActivityCreated(var1);
   }

   public void onActivityResult(int var1, int var2, Intent var3) {
      if (var1 == 203) {
         if (var2 == 1) {
            MyLog.log("MyPreferencesFragment.onActivityResult CHECK_VOICE_DATA_PASS");
            return;
         }

         MyLog.log("MyPreferencesFragment.onActivityResult CHECK_VOICE_DATA_PASS failed! No voice data!");

         try {
            var3 = new Intent();
            var3.setAction("android.speech.tts.engine.INSTALL_TTS_DATA");
            this.startActivity(var3);
            return;
         } catch (Exception var4) {
            MyLog.log("MyPreferencesFragment.onActivityResult startActivity(ACTION_INSTALL_TTS_DATA) failed!");
            MyLog.toast(this.getActivity(), "No voice data. This function is not available in your device!");
         }
      }

   }

   public void onCreate(Bundle var1) {
      MyLog.log("MyPreferencesFragment.onCreate");
      super.onCreate(var1);
      MyPreferencesActivity var5 = (MyPreferencesActivity)this.getActivity();
      Bundle var2 = this.getArguments();
      if (var2 != null) {
         String var3 = var2.getString("prefxmlres");
         String var7 = var5.getPackageName();
         this.addPreferencesFromResource(var5.getResources().getIdentifier(var3, "xml", var7));
      }

      if (App.d) {
         try {
            Intent var6 = new Intent("android.speech.tts.engine.CHECK_TTS_DATA");
            this.startActivityForResult(var6, 203);
            this.a();
            return;
         } catch (Exception var4) {
            MyLog.log("MyPreferencesFragment.onCreate ERROR in starting CHECK_TTS_DATA");
         }
      }

   }

   public void onDestroy() {
      MyLog.log("MyPreferencesFragment.onDestroy");
      App.c(this.getActivity().getApplicationContext());
      super.onDestroy();
   }

   public void onPause() {
      MyLog.log("MyPreferencesFragment.onPause");
      this.getPreferenceManager().getSharedPreferences().unregisterOnSharedPreferenceChangeListener(this);
      super.onPause();
   }

   public void onResume() {
      MyLog.log("MyPreferencesFragment.onResume");
      this.getPreferenceManager().getSharedPreferences().registerOnSharedPreferenceChangeListener(this);
      this.a(this.getPreferenceScreen());
      super.onResume();
   }

   public void onSharedPreferenceChanged(SharedPreferences var1, String var2) {
      Preference var3 = this.findPreference(var2);
      if (var3 == null) {
         StringBuilder var4 = new StringBuilder();
         var4.append("MyPreferencesFragment.onSharedPreferenceChanged: p=null key=");
         var4.append(var2);
         MyLog.log(var4.toString());
      } else {
         this.b(var3);
      }
   }

   public void onStart() {
      MyLog.log("MyPreferencesFragment.onDestroy");
      super.onStart();
   }
}

File: app/src/main/java/com/dof100/morsenotifier/MyPreferencesFragment.kt
package com.dof100.morsenotifier

import android.content.Intent
import android.content.SharedPreferences
import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import android.os.Bundle
import android.preference.CheckBoxPreference
import android.preference.Preference
import android.preference.PreferenceCategory
import android.preference.PreferenceFragment
import android.preference.PreferenceGroup
import android.preference.PreferenceManager
import android.preference.PreferenceScreen
import com.dof100.morsenotifier.MyLog.log
import com.dof100.morsenotifier.MyLog.toast
import java.util.Locale

class MyPreferencesFragment constructor() : PreferenceFragment(), OnSharedPreferenceChangeListener {
  private fun a() {
    if (App.Companion.d) {
      val var1: MyPreferenceList? = findPreference("pref_voice_locale") as MyPreferenceList?
      if (var1 == null) {
        log("MyPreferencesFragment.init_pref_voice_locale ERROR pref_voice_locale=null")
      } else {
        val var2: MyPlayerTTS? = App.Companion.a(getActivity().getApplicationContext())
        if (var2 == null) {
          log("MyPreferencesFragment.init_pref_voice_locale ERROR mPlayerTTS=null")
        } else if (var2.mTTS == null) {
          log("MyPreferencesFragment.init_pref_voice_locale ERROR App.mPlayerTTS.mTTS=null")
        } else if (var2.mTTSStatus == -1) {
          log("MyPreferencesFragment.init_pref_voice_locale ERROR mTTSStatus=ERROR")
        } else {
          val var3: Array<Locale> = Locale.getAvailableLocales()
          val var4: ArrayList<String> = ArrayList<String>()
          val var5: Int = var3.size
          for (var6 in 0 until var5) {
            val var7: Locale = var3.get(var6)
            var var8: String?
            if ((var7.toString() == "en_US_POSIX")) {
              var8 =
                "MyPreferencesFragment.init_pref_voice_locale Locale en_US_POSIX not supported"
            } else {
              val var9: Int = var2.mTTS!!.isLanguageAvailable(var7)
              if (var9 < 1) {
                continue
              }
              var4.add(var7.toString())
              var8 = "MyPreferencesFragment.init_pref_voice_locale Locale: $var7 supported $var9"
            }
            log(var8)
          }
          val var11: Array<String>
          if (var4.size > 0) {
            var11 = var4.toTypedArray() as Array<String>
          } else {
            var11 = arrayOf("(none)")
          }
          var1.setEntries(var11)
          var1.setEntryValues(var11)
          var1.a()
        }
      }
    }
  }

  private fun a(var1: Preference?) {
    if (var1 == null) {
      log("MyPreferencesFragment.checkAllPreferences: p=null")
    } else {
      if (var1 is PreferenceCategory || var1 is PreferenceScreen) {
        val var2: PreferenceGroup = var1 as PreferenceGroup
        for (var3 in var2.getPreferenceCount() - 1 downTo 0) {
          this.a(var2.getPreference(var3))
        }
      }
      this.b(var1)
    }
  }

  private fun b() {
    log("MyPreferencesFragment.check_pref_voice_locale")
  }

  private fun b(var1: Preference?) {
    if (var1 == null) {
      log("MyPreferencesFragment.checkPreference: aP = null")
    } else {
      val var2: String? = var1.getKey()
      if (var2 != null) {
        if (var2.startsWith("pref_voice_locale")) {
          this.b()
        }
        if (var2.startsWith("pref_call_contact")) {
          this.c()
        } else if (var2.startsWith("pref_sms_contact")) {
          this.d()
        } else if (var2.startsWith("pref_reminders_select")) {
          this.c(var1)
        } else if (var2.startsWith("pref_apps_filters")) {
          this.d(var1)
        } else if (var2.startsWith("pref_apps_recentnotifications")) {
          e(var1)
        } else {
          if (var1 is PreferenceCategory || var1 is PreferenceScreen) {
            val var4: String
            if (!(var2 == "pref_morse") && !(var2 == "pref_audio") && !(var2 == "pref_display")) {
              if (!(var2 == "pref_voice") || !App.Companion.c) {
                return
              }
              var4 = "MyPreferencesFragment.checkPreference: remove voice key = "
            } else {
              if (!App.Companion.d) {
                return
              }
              var4 = "MyPreferencesFragment.checkPreference: remove morse key = "
            }
            log("${var4}${var2}")
            var1.setEnabled(false)
          }
        }
      }
    }
  }

  private fun c() {
    val var1: CheckBoxPreference? =
      findPreference("pref_call_contactdisplayname") as CheckBoxPreference?
    val var2: CheckBoxPreference? =
      findPreference("pref_call_contactfirstname") as CheckBoxPreference?
    val var3: CheckBoxPreference? =
      findPreference("pref_call_contactlastname") as CheckBoxPreference?
    val var4: CheckBoxPreference? =
      findPreference("pref_call_contactinitials") as CheckBoxPreference?
    val var5: CheckBoxPreference? =
      findPreference("pref_call_contactnickname") as CheckBoxPreference?
    if ((var1 != null) && (var2 != null) && (var3 != null) && (var4 != null) && (var5 != null)) {
      val var6: Boolean
      if (!var1.isChecked() && !var2.isChecked() && !var3.isChecked() && !var4.isChecked() && !var5.isChecked()) {
        var6 = false
      } else {
        var6 = true
      }
      findPreference("pref_call_contactname_none").setEnabled(var6)
    }
  }

  private fun c(var1: Preference) {
    log("MyPreferencesFragment.check_pref_reminders_select")
    val var2: Int =
      PreferenceManager.getDefaultSharedPreferences(getActivity().getApplicationContext())
        .getInt("reminders_n", 0)
    val var3: MyPreferenceText
    val var4: String
    when (var2) {
      0 -> {
        var3 = var1 as MyPreferenceText
        var4 = "No Reminders"
      }

      1 -> {
        var3 = var1 as MyPreferenceText
        var4 = "1 reminder"
      }

      else -> {
        var3 = var1 as MyPreferenceText
        var4 = String.format(Locale.US, "%d reminders", var2)
      }
    }
    var3.setValue(var4)
  }

  private fun d() {
    val var1: CheckBoxPreference? =
      findPreference("pref_sms_contactdisplayname") as CheckBoxPreference?
    val var2: CheckBoxPreference? =
      findPreference("pref_sms_contactfirstname") as CheckBoxPreference?
    val var3: CheckBoxPreference? =
      findPreference("pref_sms_contactlastname") as CheckBoxPreference?
    val var4: CheckBoxPreference? =
      findPreference("pref_sms_contactinitials") as CheckBoxPreference?
    val var5: CheckBoxPreference? =
      findPreference("pref_sms_contactnickname") as CheckBoxPreference?
    if ((var1 != null) && (var2 != null) && (var3 != null) && (var4 != null) && (var5 != null)) {
      val var6: Boolean
      if (!var1.isChecked() && !var2.isChecked() && !var3.isChecked() && !var4.isChecked() && !var5.isChecked()) {
        var6 = false
      } else {
        var6 = true
      }
      findPreference("pref_sms_contactname_none").setEnabled(var6)
    }
  }

  private fun d(var1: Preference) {
    log("MyPreferencesFragment.check_pref_apps_filters")
    var var2: Int = MyAppNotificationFilters.Companion.c(getActivity().getApplicationContext())
    val var3: String
    val var4: MyPreferenceText
    if (var2 > 1) {
      var4 = var1 as MyPreferenceText
      var3 =
        String.format(Locale.US, this.getString(R.string.pref_apps_filters_dfilters), var2)
    } else {
      if (var2 == 1) {
        var4 = var1 as MyPreferenceText
        var2 = R.string.pref_apps_filters_1filter
      } else {
        var4 = var1 as MyPreferenceText
        var2 = R.string.pref_apps_filters_nofilters
      }
      var3 = this.getString(var2)
    }
    var4.setValue(var3)
  }

  private fun e(var1: Preference) {
    log("MyPreferencesFragment.check_pref_apps_recentnotifications")
    val var2: Int = MyRecentAppNotifications.Companion.n(getActivity().getApplicationContext())
    val var3: MyPreferenceText
    val var4: String
    if (var2 > 1) {
      var3 = var1 as MyPreferenceText
      var4 = String.format(Locale.US, "%d recent notifications", var2)
    } else if (var2 == 1) {
      var3 = var1 as MyPreferenceText
      var4 = "1 recent notification"
    } else {
      var3 = var1 as MyPreferenceText
      var4 = "No recent notifications"
    }
    var3.setValue(var4)
  }

  public override fun onActivityCreated(var1: Bundle?) {
    log("MyPreferencesFragment.onActivityCreated")
    super.onActivityCreated(var1)
  }

  public override fun onActivityResult(var1: Int, var2: Int, var3: Intent) {
    var var3: Intent = var3
    if (var1 == 203) {
      if (var2 == 1) {
        log("MyPreferencesFragment.onActivityResult CHECK_VOICE_DATA_PASS")
        return
      }
      log("MyPreferencesFragment.onActivityResult CHECK_VOICE_DATA_PASS failed! No voice data!")
      try {
        var3 = Intent()
        var3.setAction("android.speech.tts.engine.INSTALL_TTS_DATA")
        this.startActivity(var3)
        return
      } catch (var4: Exception) {
        log("MyPreferencesFragment.onActivityResult startActivity(ACTION_INSTALL_TTS_DATA) failed!")
        toast(
          getActivity(),
          "No voice data. This function is not available in your device!"
        )
      }
    }
  }

  public override fun onCreate(var1: Bundle?) {
    log("MyPreferencesFragment.onCreate")
    super.onCreate(var1)
    val var5: MyPreferencesActivity = getActivity() as MyPreferencesActivity
    val var2: Bundle? = getArguments()
    if (var2 != null) {
      val var3: String? = var2.getString("prefxmlres")
      val var7: String = var5.getPackageName()
      addPreferencesFromResource(var5.getResources().getIdentifier(var3, "xml", var7))
    }
    if (App.Companion.d) {
      try {
        val var6: Intent = Intent("android.speech.tts.engine.CHECK_TTS_DATA")
        this.startActivityForResult(var6, 203)
        this.a()
        return
      } catch (var4: Exception) {
        log("MyPreferencesFragment.onCreate ERROR in starting CHECK_TTS_DATA")
      }
    }
  }

  public override fun onDestroy() {
    log("MyPreferencesFragment.onDestroy")
    App.Companion.c(getActivity().getApplicationContext())
    super.onDestroy()
  }

  public override fun onPause() {
    log("MyPreferencesFragment.onPause")
    getPreferenceManager().getSharedPreferences()
      .unregisterOnSharedPreferenceChangeListener(this)
    super.onPause()
  }

  public override fun onResume() {
    log("MyPreferencesFragment.onResume")
    getPreferenceManager().getSharedPreferences().registerOnSharedPreferenceChangeListener(this)
    this.a(getPreferenceScreen())
    super.onResume()
  }

  public override fun onSharedPreferenceChanged(var1: SharedPreferences, var2: String?) {
    val var3: Preference? = findPreference(var2)
    if (var3 == null) {
      log("MyPreferencesFragment.onSharedPreferenceChanged: p=null key=$var2")
    } else {
      this.b(var3)
    }
  }

  public override fun onStart() {
    log("MyPreferencesFragment.onDestroy")
    super.onStart()
  }
}
